### PR TITLE
[REF] Clean up feature functions

### PR DESCRIPTION
**Pull request by tsalo**
Closes None.

Changes proposed in this pull request:

- Remove unused arguments in `feature_spatial`.
- Simplify value calculation in `feature_spatial`.
- Clean up `feature_frequency`.
- Improve docstrings for all three feature functions.

### CODE REVIEW

File: aroma/features.py [hunk 1]
```diff
@@ -123,10 +129,11 @@ def feature_frequency(melFTmix, TR):
 
     # Load melodic_FTmix file
     FT = np.loadtxt(melFTmix)
+    n_frequencies, _ = FT.shape
```
Comment (eurunuela) [inline aroma/features.py:132]: Why not `FT.shape[0]`?

File: aroma/features.py [hunk 2]
```diff
@@ -198,55 +202,26 @@ def feature_spatial(fslDir, tempDir, aromaDir, melIC):
...
-        if not (outVox == 0):
-            outMean = np.mean(out_data[out_data != 0])
-        else:
-            outMean = 0
-
-        outSum = outMean * outVox
+        outSum = np.sum(out_data)
 
         # Determine edge and CSF fraction
         if not (totSum == 0):
```
Comment (eurunuela) [inline aroma/features.py:227]: Why not `if (totSum != 0):`?

File: aroma/features.py [hunk 3]
```diff
@@ -198,55 +202,26 @@ def feature_spatial(fslDir, tempDir, aromaDir, melIC):
         # Get sum of Z-values within the total Z-map (calculate via the mean
         # and number of non-zero voxels)
         tempICdata = tempIC.get_fdata()
-        totVox = np.sum(tempICdata != 0)  # number of nonzero voxels in image
+        totSum = np.sum(tempICdata)
```
Comment (eurunuela) [inline aroma/features.py:205]: They count the number of non-zero voxels while you're suggesting we sum the Z-values (as the comment says). Do we know which one is correct? 

Evey other sum variable is also calculated by counting the number of non-zero values.

**Review by eurunuela — CHANGES_REQUESTED** [review]
I have a doubt regarding the calculation of `totSum`.

File: aroma/features.py (hunk 3 above)
Comment (tsalo) [reply aroma/features.py:205]: They calculate the sum by counting the non-zero voxels, then calculating the mean across non-zero voxels, and the multiplying the number of non-zero voxels by the mean of the non-zero voxels. The results should be the same as just summing all voxels.

File: aroma/features.py (hunk 2 above)
Comment (tsalo) [reply aroma/features.py:227]: good catch

File: aroma/features.py (hunk 1 above)
Comment (tsalo) [reply aroma/features.py:132]: I think it's a holdover from when I thought the number of components might come into play in this function. I'll change it.

File: aroma/features.py (hunk 3 above)
Comment (eurunuela) [reply aroma/features.py:205]: Then why don't we just sum all non-zero voxels? It sounds simpler to me.

Comment (tsalo) [reply aroma/features.py:205]: Zeros don't contribute to a sum though, so filtering the voxels based on that doesn't end up changing anything. I also think that `np.sum(tempICdata)` is simpler than `np.sum(tempICdata[tempICdata != 0])`.

Comment (eurunuela) [reply aroma/features.py:205]: I understand now, thanks!

**Review by eurunuela — APPROVED** [review]
Thank you! LGTM!

**Pull request merged**